### PR TITLE
Update Travis Rubies

### PR DESCRIPTION
**Pull request by bdewater**
Ruby 2.4 ships with OpenSSL gem 2.0, make sure everything still works.

### CODE REVIEW

**Comment by bdewater** [comment]
Something's weird going on 🤔 and I've hidden it behind a test helper...

Ruby 2.3.1:
```
request = Acme::Client::CertificateRequest.new(names: %w(example.org www.example.org), private_key: test_key)
request.csr.attributes.first.value.first.first.value.first.value
=> "subjectAltName"
```
Ruby 2.4.0:
```
request = Acme::Client::CertificateRequest.new(names: %w(example.org www.example.org), private_key: test_key)
request.csr.attributes.first.value.value.first.value.first.value.first.value
=> "subjectAltName"
```

File: spec/certificate_request_spec.rb [hunk 1]
```diff
@@ -130,10 +130,10 @@
     request = Acme::Client::CertificateRequest.new(names: %w(example.org www.example.org), private_key: test_key)
 
     extension = request.csr.attributes.find { |attribute|
-      attribute.value.first.first.value.first.value == 'subjectAltName'
```
Comment (unixcharles) [inline spec/certificate_request_spec.rb]: Such a weird API change. Do you happen to know what lead to that change? I couldn't find anything in the openssl change log / repo.

File: spec/support/asn1_helper.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module Asn1Helper
+  def asn1_dig(obj)
+    if obj.respond_to?(:first) && obj.first.is_a?(Enumerable)
+      asn1_dig obj.first
+    elsif obj.respond_to?(:value) && obj.value.is_a?(Enumerable)
+      asn1_dig obj.value
+    else
+      obj
+    end
```
Comment (unixcharles) [inline spec/support/asn1_helper.rb]: I wonder if we shouldn't just specifically do conditional assertion on `OpenSSL::VERSION`? What do you think?

Comment (bdewater) [inline spec/support/asn1_helper.rb]: Yeah, it's definitely more clear but might break again somewhere in the future. Then again, ASN1 is always a PITA to work with so let's deal with it when the problem arises 😀

Comment (unixcharles) [inline spec/support/asn1_helper.rb]: I would personally prefer it to fail. I'm scare this could unexpectedly work, which would be worst 😄

**Comment by unixcharles** [comment]
Thanks Bart!

**Pull request merged**